src/link/MachO.zig

```diff
@@ -702,17 +702,23 @@ pub fn flushModule(self: *MachO, comp: *Compilation, prog_node: *std.Progress.No
     if (self.dyld_stub_binder_index == null) {
         self.dyld_stub_binder_index = try self.addUndefined("dyld_stub_binder", .add_got);
     }
-    // if (!self.base.options.single_threaded) {
-    //     _ = try self.addUndefined("_tlv_bootstrap", .none);
-    // }
+    if (!self.base.options.single_threaded) {
+        _ = try self.addUndefined("__tlv_bootstrap", .none);
+    }
 
     try self.createMhExecuteHeaderSymbol();
 
     var actions = std.ArrayList(ResolveAction).init(self.base.allocator);
     defer actions.deinit();
     try self.resolveSymbolsInDylibs(&actions);
 
-    if (self.unresolved.count() > 0) {
+    blk: {
+        for (self.unresolved.keys()) |index| {
+            // TODO: convert to compiler errors.
+            const global = self.globals.items[index];
+            const sym_name = self.getSymbolName(global);
+            log.err("undefined symbol reference '{s}'", .{sym_name});
+        } else break :blk;
         return error.UndefinedSymbolReference;
```

Comment:
There are no breaks out of the for loop, so the for else always runs and this return is unreachable.

Reply:
Oh I see it now, yes, you are right. Thanks for pointing the one!